rotkehlchen/db/updates.py

```diff
@@ -166,6 +172,63 @@ def update_rpc_nodes(self, data: list[dict[str, Any]], version: int) -> None:
             new_default_nodes=new_default_nodes,
         )
 
+    def update_accounting_rules(self, data: list[dict[str, Any]], version: int) -> None:
+        log.info(f'Applying update for accounting rules to v{version}')
+        rules_db = DBAccountingRules(self.user_db)
+        for rule_data in data:
+            try:
+                event_type = HistoryEventType.deserialize(rule_data['event_type'])
+                event_subtype = HistoryEventSubType.deserialize(rule_data['event_subtype'])
+                counterparty = rule_data['counterparty'] if rule_data['counterparty'] is not None else NO_ACCOUNTING_COUNTERPARTY  # noqa: E501
+                rule = BaseEventSettings(
+                    taxable=rule_data['taxable'],
+                    count_entire_amount_spend=rule_data['count_entire_amount_spend'],
+                    count_cost_basis_pnl=rule_data['count_cost_basis_pnl'],
+                    method=rule_data['method'],
+                    accounting_treatment=TxAccountingTreatment.deserialize(rule_data['accounting_treatment']) if rule_data['accounting_treatment'] else None,  # noqa: E501
+                )
+            except KeyError as e:
+                log.error(f'Failed to read key {e} while iterating new accounting rules')
+
+            try:
+                rule_id = rules_db.add_accounting_rule(
+                    event_type=event_type,
+                    event_subtype=event_subtype,
+                    counterparty=counterparty,
+                    rule=rule,
+                )
+            except InputError as e:
+                # there is a conflict in the rule. Notify the frontend about it
+                rules, _ = rules_db.query_rules_and_serialize(
+                    filter_query=AccountingRulesFilterQuery.make(
+                        event_types=[event_type],
+                        event_subtypes=[event_subtype],
+                        counterparties=[counterparty],
+                    ),
+                )
+                if len(rules) == 1:
```

Comment:
If it's > 1 what happens? When can that happen?

Reply:
it can be either 0 or 1 since the combination is unique